packages/room/peer/peer.js

```diff
@@ -646,6 +655,25 @@ export const createPeer = ({ api, createStream, event, streams, config }) => {
 
       this._event.emit(PeerEvents.STREAM_AVAILABLE, { stream })
     }
+
+    /**
+     * @param {import('./peer-types.d.ts').RoomPeerType.VoiceActivity} vad
+     */
+    _onVoiceActivity = (vad) => {
+      const event = new CustomEvent('voiceactivity', {
+        detail: {
+          voiceActivity: vad,
+        },
+      })
+
+      this.dispatchEvent(event)
```

Comment:
1. Because this is a custom event, and the usage is different from the current event handling implementation, I think we need to add the usage information in the readme.

2. Also, I want to ask about the current event emitter implementation using `on` method mas. Should we replace the `on` event mechanism and using CustomEvent approach similar to this one? From what I see when using the CustomEvent, users will not be able to have auto suggestion for event name. The auto suggestion for sdk is especially helpful when we haven't provided the users enough information in the readme about the list of available events.

Reply:
1. Yes, I will add it to readme.
2. Agree, I think we should add the same approach for `room.on` to make it closer to native. Other language can use their own event listener style to make it familiar with the developers with that language.

Reply:
Okay mas. I will remove `room.on` and use `addEventListener` for the future implementation